From 56c46018dc0e9890dc958124650a2aa2cfdf5e3c Mon Sep 17 00:00:00 2001
From: jzhuan5 <jin.can.zhuang@intel.com>
Date: Tue, 3 Sep 2013 19:48:49 -0400
Subject: libc: functionfs add ss_count to support superspeed

BZ: 135849

Add ss_count to support superspeed

Change-Id: I4462126b11dc89743b0838c27b801b0fe72b815a
Category: aosp improvement
Domain: usb adb
Origin: internal
Upstream-Candidate: no
Signed-off-by: jzhuan5 <jin.can.zhuang@intel.com>
Reviewed-on: http://android.intel.com:8080/129456
Reviewed-by: Gross, Mark <mark.gross@intel.com>
Reviewed-by: Chen, HuaqiangX <huaqiangx.chen@intel.com>
Reviewed-by: Yao, Yong <yong.yao@intel.com>
Tested-by: Li, GuangleiX <guangleix.li@intel.com>
Tested-by: Wang, Yu Y <yu.y.wang@intel.com>
Reviewed-by: cactus <cactus@intel.com>
Tested-by: cactus <cactus@intel.com>
---
 libc/kernel/common/linux/usb/functionfs.h |   16 +++++++++-------
 1 files changed, 9 insertions(+), 7 deletions(-)

diff --git a/libc/kernel/common/linux/usb/functionfs.h b/libc/kernel/common/linux/usb/functionfs.h
index 11e8d4c..4807cca 100644
--- a/libc/kernel/common/linux/usb/functionfs.h
+++ b/libc/kernel/common/linux/usb/functionfs.h
@@ -43,39 +43,41 @@ struct usb_functionfs_descs_head {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
  __le32 fs_count;
  __le32 hs_count;
+ __le32 ss_count;
 } __attribute__((packed));
-struct usb_functionfs_strings_head {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+struct usb_functionfs_strings_head {
  __le32 magic;
  __le32 length;
  __le32 str_count;
- __le32 lang_count;
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+ __le32 lang_count;
 } __attribute__((packed));
 enum usb_functionfs_event_type {
  FUNCTIONFS_BIND,
- FUNCTIONFS_UNBIND,
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+ FUNCTIONFS_UNBIND,
  FUNCTIONFS_ENABLE,
  FUNCTIONFS_DISABLE,
  FUNCTIONFS_SETUP,
- FUNCTIONFS_SUSPEND,
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+ FUNCTIONFS_SUSPEND,
  FUNCTIONFS_RESUME
 };
 struct usb_functionfs_event {
- union {
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+ union {
  struct usb_ctrlrequest setup;
  } __attribute__((packed)) u;
  __u8 type;
- __u8 _pad[3];
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+ __u8 _pad[3];
 } __attribute__((packed));
 #define FUNCTIONFS_FIFO_STATUS _IO('g', 1)
 #define FUNCTIONFS_FIFO_FLUSH _IO('g', 2)
-#define FUNCTIONFS_CLEAR_HALT _IO('g', 3)
 /* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
+#define FUNCTIONFS_CLEAR_HALT _IO('g', 3)
 #define FUNCTIONFS_INTERFACE_REVMAP _IO('g', 128)
 #define FUNCTIONFS_ENDPOINT_REVMAP _IO('g', 129)
 #endif
+/* WARNING: DO NOT EDIT, AUTO-GENERATED CODE - SEE TOP FOR INSTRUCTIONS */
-- 
1.7.4.1

